lib/base-compiler.js

```diff
@@ -408,6 +484,17 @@ class BaseCompiler {
                             return result;
                         })
                         .then(result => {
+                            if (doExecute) {
```

Comment:
I'm not sure whether we should cache first, execute later? Or cache while we execute. Seems like we shouldn't wait to put this in the cache until execution works. Also, we probably want to cache the compilation results even if the execute fails? Again, not sure here, don't think there's a right answer, but wondered if you had any thoughts?

Reply:
I think the most ideal thing would be if executables would be cached as early as possible.

For example when some compilers actually need to produce executables to see the asm, or when someone clicks binary mode and it produces an executable (some only produce object files) - if we cache it then and there, you could get it from the cache without recompiling when someone hits execute.

But when execution happens I think we should probably cache the execution result as well maybe? But then we can't cleanly support things like time or rand() related executables (unless the commandline parameters are changed in between tries).

Reply:
Wait, I understand what you mean now, I think.
I will see and test how early I can call it...

Reply:
Changed some things around, but kept the waiting for execution result. Need to think about it some more

Reply:
removed the execResult from the caching